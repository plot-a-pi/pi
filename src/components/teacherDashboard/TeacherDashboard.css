.TeacherDashboard {
    display: flex;
    flex-direction: column;
    width: 95vw;
    margin: auto;

    .instructions {
        padding: 1em;
        margin-top: 3vh;
        margin-bottom: 2vh;
        box-shadow: 0 1px 0 3px rgba(0,0,0,.25);
    }

    .loginButton {
        display: flex;
        justify-content: center;
        height: 30px;
    }
}