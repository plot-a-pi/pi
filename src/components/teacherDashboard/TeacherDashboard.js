import React, { useEffect } from 'react';
import { useUser } from '../../firebase/AuthProvider';
import { useHistory } from 'react-router-dom';
import { loginWithProvider } from '../../firebase/firebase';
import styles from './TeacherDashboard.css';

const TeacherDashboard = () => {
  const user = useUser();
  const history = useHistory();

  useEffect(() => {
    if(user) history.replace('/teacher-sessions');
  }, [user]);

  const handleClick = () => {
    loginWithProvider();
  };

  return (
    <>
      <div className={styles.TeacherDashboard}>
        <div className={styles.instructions}>
          <h2>Instructions</h2>
          <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
        </div>
        <div className={styles.loginButton}>
          <button onClick={handleClick}>SignUp/Login</button>
        </div>
      </div>
    </>
  );
};

export default TeacherDashboard;
