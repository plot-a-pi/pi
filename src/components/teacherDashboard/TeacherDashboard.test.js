import React from 'react';
import { shallow } from 'enzyme';
import TeacherDashboard from './TeacherDashboard';

jest.mock('../../firebase/AuthProvider', () => ({
  useUser: () => {}, 
}));

jest.mock('react-router-dom', () => ({
  useHistory: () => {}, 
}));

describe('TeacherDashboard component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<TeacherDashboard />);
    expect(wrapper).toMatchSnapshot();
  });
});
