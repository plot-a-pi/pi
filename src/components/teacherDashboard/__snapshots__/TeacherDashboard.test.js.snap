// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherDashboard component matches a snapshot 1`] = `
<Fragment>
  <div
    className="TeacherDashboard"
  >
    <div
      className="loginButton"
    >
      <button
        onClick={[Function]}
      >
        SignUp/Login
      </button>
    </div>
  </div>
</Fragment>
`;
